src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,243 @@
+///<reference path="../../reference.ts" />
+
+module Plottable {
+export module Component {
+  export module Interactive {
+    type _EdgeIndicator = {
+      top: boolean;
+      bottom: boolean;
+      left: boolean;
+      right: boolean;
+    }
+
+    export class DragBoxLayer extends Component.SelectionBoxLayer {
+      private _dragInteraction: Interaction.Drag;
+      private _boxEdgeT: D3.Selection;
+      private _boxEdgeB: D3.Selection;
+      private _boxEdgeL: D3.Selection;
+      private _boxEdgeR: D3.Selection;
+      private _boxCornerTL: D3.Selection;
+      private _boxCornerTR: D3.Selection;
+      private _boxCornerBL: D3.Selection;
+      private _boxCornerBR: D3.Selection;
+
+      private _detectionRadius = 3;
+      private _xResizable = false;
+      private _yResizable = false;
+
+      constructor() {
+        super();
+        this.classed("drag-box-layer", true);
+
+        this._dragInteraction = new Interaction.Drag();
+
+        var grabbedEdges: _EdgeIndicator;
+        var _initTL: Point;
+        var _initBR: Point;
+
+        this._dragInteraction.onDragStart((s: Point) => {
+          grabbedEdges = this._getEdges(s);
+          if (!this._yResizable) {
+            grabbedEdges.top = false;
+            grabbedEdges.bottom = false;
+          }
+          if (!this._xResizable) {
+            grabbedEdges.left = false;
+            grabbedEdges.right = false;
+          }
+
+          if (!this.boxVisible() ||
+              (!grabbedEdges.top && !grabbedEdges.bottom &&
+               !grabbedEdges.left && !grabbedEdges.right)
+             ) {
+            this.bounds({
+              topLeft: s,
+              bottomRight: s
+            });
+            // a new box is effectively "grabbed" at the bottom-right corner
+            grabbedEdges.bottom = true;
+            grabbedEdges.right = true;
+          }
+
+          this.boxVisible(true);
+          var bounds = this.bounds();
+          _initTL = bounds.topLeft;
+          _initBR = bounds.bottomRight;
```

Comment:
Is _initTL and _initBR different from bounds.topLeft/bottomRight at any sensible time?